js/virtual-keyboard.js

```diff
@@ -0,0 +1,213 @@
+    this.shiftPressed = false;
+    this.keepPressedButtonsArray = [];
+  }
+
+  createDomKeyboard() {
+    const { body } = document;
+    const keyboard = document.createElement('div');
+    body.appendChild(keyboard);
+    keyboard.classList.add('virtual-keyboard');
+    const keyArr = createVirtualKeyboardKeys();
+
+    for (let i = 1; i < 6; i += 1) {
+      const domKeyboardRow = document.createElement('div');
+      domKeyboardRow.id = `row_${i}`;
+      domKeyboardRow.classList.add('keyboard__row');
+      keyboard.appendChild(domKeyboardRow);
+    }
+
+    for (let i = 0; i < keyArr.length; i += 1) {
+      const kKey = keyArr[i];
+      const el = document.createElement('div');
+      el.innerHTML = this.englishLanguage ? kKey.valueLabel : kKey.rusValueLabel;
+      el.style.width = `${kKey.width}px`;
+      el.classList.add('keyboard__key');
+      el.id = kKey.id;
+      document.getElementById(`row_${kKey.row}`).appendChild(el);
+    }
+
+    return document.querySelectorAll('.virtual-keyboard div');
+  }
+
+  addActiveKeyboardKey() {
+    this.domKeys.forEach((e) => {
+      e.addEventListener('mousedown', (event) => {
+        // CAPS LOCK
+        if (event.target.id !== '20') {
+          event.target.classList.add(this.cssClassNameConfig.keyboardKeyActive);
+        }
+        this.pressKey(event);
+      });
+      e.addEventListener('mouseup', () => {
+        this.domKeys.forEach((el) => this.removeClassKeyActive(el));
+      });
+    });
+  }
+
+  removeClassKeyActive(domKey) {
+    domKey.classList.remove(this.cssClassNameConfig.keyboardKeyActive);
+  }
+
+  bindPhysicalKeyboardEvents() {
+    const domKeys = [...this.domKeys];
+
+    document.addEventListener('keydown', (event) => {
+      const activeKey = domKeys.find((e) => +e.id === +event.keyCode);
+      if (typeof activeKey !== 'undefined') {
+        if (activeKey.id !== '20') {
+          activeKey.classList.add(this.cssClassNameConfig.keyboardKeyActive);
+        }
+        this.addKeepPressedButtonsArray(activeKey.id);
+        this.pressKey(event);
+      }
+      event.preventDefault();
+    });
+
+    document.addEventListener('keyup', (event) => {
+      const activeKey = domKeys.find((e) => +e.id === +event.keyCode);
+      if (typeof activeKey !== 'undefined') {
+        domKeys.forEach((el) => this.removeClassKeyActive(el));
+        this.removeKeepPressedButtonsArray(activeKey.id);
+      }
+    });
+  }
+
+  addKeepPressedButtonsArray(id) {
+    if (this.keepPressedButtonsArray.indexOf(id) === -1) {
+      this.keepPressedButtonsArray.push(id);
+    }
+  }
+
+  removeKeepPressedButtonsArray(id) {
+    const index = this.keepPressedButtonsArray.indexOf(id);
+
+    this.keepPressedButtonsArray.splice(index, 1);
+  }
+
+  toggleCapsLock() {
+    this.capsLockEnabled = !this.capsLockEnabled;
+    document.getElementById('20').classList.toggle('keyboard__caps-lock_active');
+  }
+
+  prepareSymbolToShowInScreen(vKey) {
+    let result = '';
```

Comment:
It isn't good structure. It is very hard to read

Reply:
I fully agree with You,
but 14 conditions are checked here, and I didn’t come up with anything better